.github/workflows/build-metrics.yml

```diff
@@ -116,3 +116,54 @@ jobs:
             --pr_number "${{ github.event.number }}" \
             --sha "${{ inputs.ref || github.sha }}" \
             "/tmp/metrics"
+
+  upload-report:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    name: Generate and Upload Build Metric Report
+    needs: metrics
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      - name: Setup Git User
+        run: |
+          git config --global user.email "velox@users.noreply.github.com"
+          git config --global user.name "velox"
+
+      - uses: cachix/install-nix-action@8887e596b4ee1134dae06b98d573bd674693f47c # v26
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      - name: Build Environment
+        run: |
+          cd scripts/bm-report
+          nix-build
+
+      - name: Build Documentation
+        env:
+          CONBENCH_URL: "https://velox-conbench.voltrondata.run/"
+          CONBENCH_EMAIL: "${{ secrets.CONBENCH_EMAIL }}"
+          CONBENCH_PASSWORD: "${{ secrets.CONBENCH_PASSWORD }}"
+        run: |
+          cd scripts/bm-report
+          nix-shell --run "quarto render report.qmd"
+
+      - name: Push Report
+        # The report only uses conbench data from 'main'
+        # so any data generated in a PR won't be included
+        if: ${{ github.event_name != 'pull_request' && github.repository == 'facebookincubator/velox'}}
+        run: |
+          git checkout gh-pages
+          mkdir -p docs/bm-report
+          cp -R scripts/bm-report/report.html docs/bm-report/index.html
```

Comment:
Does this overwrite say the previous report ?

Reply:
Yes, as long as the filename is the same.